### PR TITLE
[SU-74] Add yarn SDK for VS Code

### PR DESCRIPTION
**Pull request by nawatts**
To get the VS Code ESLint plugin to work after #2906, I had to run `yarn dlx "@yarnpkg/sdks" vscode`. This adds the resulting yarn files and VS Code configuration.

https://next.yarnpkg.com/advanced/pnpify#ide-support
https://next.yarnpkg.com/getting-started/editor-sdks

If we would rather not add configuration for specific editors, can we ignore `.yarn/sdks`?

### CODE REVIEW

**Review by zarsky-broad — APPROVED** [review]
It's explicitly unignored by yarn's default gitignore setup, so I say we commit it. What's the zipfs extension, though?

**Comment by nawatts** [comment]
> What's the zipfs extension, though?

It lets you mount zip files as a folder in the VS Code workspace and browse their contents. It's intended for use with the Yarn cache. It may be useful if you need to look through some library code.

https://marketplace.visualstudio.com/items?itemName=arcanis.vscode-zipfs

File: .yarn/sdks/eslint/bin/eslint.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+const absPnpApiPath = resolve(__dirname, relPnpApiPath);
+const absRequire = (createRequire || createRequireFromPath)(absPnpApiPath);
+
+if (existsSync(absPnpApiPath)) {
+  if (!process.versions.pnp) {
+    // Setup the environment to be able to require eslint/bin/eslint.js
+    require(absPnpApiPath).setup();
+  }
+}
```
Comment (kyuksel) [inline .yarn/sdks/eslint/bin/eslint.js:17]: Consider

```suggestion
if (existsSync(absPnpApiPath) && !process.versions.pnp) {
  // Setup the environment to be able to require eslint/bin/eslint.js
  require(absPnpApiPath).setup();
}
```

Similarly in the next file

**Review by kyuksel — APPROVED** [review]
I'm not familiar with VS Code but the changes LGTM with a minor comment inline.

File: .yarn/sdks/eslint/bin/eslint.js (hunk 1 above)
Comment (zarsky-broad) [reply .yarn/sdks/eslint/bin/eslint.js:17]: These files are generated, unless I'm mistaken.

Comment (nawatts) [reply .yarn/sdks/eslint/bin/eslint.js:17]: Yes, they were generated by yarn.

**Pull request merged**